TezosKit/Client/TezosNodeClient.swift

```diff
@@ -328,29 +331,41 @@ public class TezosNodeClient: AbstractClient {
   public func runOperation(
     _ operation: Operation,
     from wallet: Wallet,
-    completion: @escaping ([String: Any]?, Error?) -> Void
+    completion: @escaping (Result<[String: Any], TezosKitError>) -> Void
   ) {
-    getMetadataForOperation(address: wallet.address) { [weak self] operationMetadata in
-      guard let self = self,
-            let metadata = operationMetadata else {
-        completion(nil, nil)
+    getMetadataForOperation(address: wallet.address) { [weak self] result in
+      guard let self = self else {
         return
       }
-      let operationPayload = self.createOperationPayload(operations: [operation], operationMetadata: metadata)
-      self.forgeOperation(operationPayload: operationPayload, operationMetadata: metadata) { [weak self] bytes, error in
-        guard let self = self,
-              let bytes = bytes,
-              let (_, signedOperationPayload) = self.sign(
-                operationPayload: operationPayload,
-                forgedPayload: bytes,
-                keys: wallet.keys
-              ) else {
-            let error = TezosKitError(kind: .unknown, underlyingError: nil)
-            completion(nil, error)
+
+      switch result {
+      case .failure(let error):
+        completion(.failure(error))
+        return
+      case .success(let metadata):
+        let operationPayload = self.createOperationPayload(operations: [operation], operationMetadata: metadata)
+        self.forgeOperation(operationPayload: operationPayload, operationMetadata: metadata) { [weak self] result in
```

Comment:
too much nesting, move to a helper method

Reply:
Done.